src/server/pkg/storage/fileset/storage.go

```diff
@@ -166,7 +166,22 @@ type CompactSpec struct {
 // CompactSpec returns a compaction specification that determines the input filesets (the diff file set and potentially
 // compacted filesets) and output fileset.
 func (s *Storage) CompactSpec(ctx context.Context, fileSet string, compactedFileSet ...string) (*CompactSpec, error) {
+	if len(compactedFileSet) > 1 {
+		return nil, errors.WithStack(errors.Errorf("multiple compacted FileSets"))
```

Comment:
It looks like `Errorf` grabs the stack, so wrapping it with `WithStack` should be unnecessary.